### PR TITLE
Feat/game phases

### PR DESCRIPTION
**Pull request by naikibro**
## Changes
- created all required pages
- implemented the passing of the name from page to page
- the players now exist
- created a beautifull scoreboard
- created a player panel to show the player image

## TODO
- implement the `GameOverPage`
- handle the static global variables better as context ( turn, phase, scores... )
- merge correctly

REF: #20

### CODE REVIEW

**Comment by naikibro** [comment]
[Screencast from 05-14-2024 10:55:58 AM.webm](https://github.com/naikibro/BatailleNavale/assets/64978347/efdc0159-1439-44d8-a90e-fea297337d5f)

File: gui/src/common/Game.java [hunk 1]
```diff
@@ -2,20 +2,18 @@
...
+    public Game(Map map){
 
-        this.map = map;
-        this.isPlaying = false;
-        this.phase = 0;
+        Game.map = map;
+        Game.isPlaying = false;
+        Game.phase = 0;
+        Game.turn = 0;
     }
```
Comment (tteuru09) [inline gui/src/common/Game.java:24]: C'est pas mieux de mettre "this.map" au lieu de "Game.map" ?

**Review by tteuru09 — COMMENTED** [review]
Seems good to me ! except for the little comment i left.

**Review by tteuru09 — COMMENTED** [review]
Seems good to me

**Pull request merged**